drivers/accel/adxl355/adxl355.c

```diff
@@ -612,10 +630,22 @@ int adxl355_get_temp(struct adxl355_dev *dev, struct adxl355_frac_repr *temp)
 	int ret;
 
 	ret = adxl355_get_raw_temp(dev, &raw_temp);
-	if (!ret)
-		temp->integer = no_os_div_s64_rem(adxl355_temp_conv(raw_temp),
-						  ADXL355_TEMP_OFFSET_DIV*ADXL355_TEMP_SCALE_FACTOR_DIV,
-						  &(temp->fractional));
+	if (!ret) {
+		switch(dev->dev_type) {
```

Comment:
This switch has almost identical code in its branches. I think duplicating code like that results in higher code size in .text and we should avoid it. One idea is to create a variable for that one parameter that is different, and call the function after the switch.